src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -484,6 +484,40 @@ async Task<DurableOrchestrationStatus> IDurableOrchestrationClient.GetStatusAsyn
             return await GetDurableOrchestrationStatusAsync(state, this.client, showHistory, showHistoryOutput);
         }
 
+        /// <inheritdoc />
+        async Task<IList<DurableOrchestrationStatus>> IDurableOrchestrationClient.GetStatusAsync(IEnumerable<string> instanceID, bool showHistory, bool showHistoryOutput, bool showInput)
+        {
+            IList<DurableOrchestrationStatus> durableorchestrationstateList = new List<DurableOrchestrationStatus>();
+            foreach (string id in instanceID)
+            {
+                IList<OrchestrationState> stateList = null;
+                try
+                {
+                    stateList = await this.DurabilityProvider.GetOrchestrationStateWithInputsAsync(id, showInput);
+                }
+                catch (NotImplementedException)
+                {
+                    // TODO: Ignore the show input flag. Should consider logging a warning.
+                    stateList = await this.client.ServiceClient.GetOrchestrationStateAsync(id, allExecutions: false);
+                }
+
+                OrchestrationState state = stateList?.FirstOrDefault();
+                DurableOrchestrationStatus currentidstate = new DurableOrchestrationStatus();
+                if (state == null || state.OrchestrationInstance == null)
+                {
+                    currentidstate.InstanceId = id;
```

Comment:
I'm not sure it's a good idea to return results for orchestration instances that don't exist. Callers may not be able to distinguish between existing and non-existing instances. It may be preferable to instead add a `null` result to the output rather than adding an incomplete result to the output.